### ISSUE TITLE
Feature to set secret for each value in KongPlugin/KongClusterPlugin

### ISSUE THREAD

**Issue by mflendrich** [Kong/kubernetes-ingress-controller#2098]
Make it possible to pull only a single value from a plugin setting from a secret, as opposed to the whole configuration being pulled.

Here is an ideal example. Currently not supported.
```
apiVersion: configuration.konghq.com/v1
kind: KongPlugin
metadata:
  name: correlation-id-demo
  namespace: echo
  annotations:
    kubernetes.io/ingress.class: kong 
  labels:
config:
  header_name:
    valueFrom:
      secretKeyRef:
        name: kong-config
        key: headername
```

Proposed solutions:
- Create a new api version and make a breaking change to the config/configFrom field
- Add a new field (mutually exclusive with what is there today to specify config) to the existing apiversion of Kong(Cluster)Plugin

**Comment by kxs-sindrakumar** [comment]
@mflendrich have you made any progress on this? I am looking at configFrom, but can't get it to work, any light on the format of configFrom you can shed?

**Comment by mflendrich** [comment]
Closing as "won't fix" because of (relative) complexity, possible interaction with secret management work happening in parallel in Kong Gateway, and little demand for this feature.

Please feel free to comment if interested in this.

@kxs-sindrakumar if you intend to use `configFrom` in its current form (that is, the entire plugin config being pulled from a kubernetes `Secret`), the usage reference [is in the docs](https://docs.konghq.com/kubernetes-ingress-controller/2.0.x/references/custom-resources/#kongplugin).
